File: TransCoder_tokenized_test_set_functions/cpp/SEARCH_IN_ROW_WISE_AND_COLUMN_WISE_SORTED_MATRIX.cpp
int search ( int mat [ 4 ] [ 4 ], int n, int x ) {
  if ( n == 0 ) return - 1;
  int smallest = a [ 0 ] [ 0 ], largest = a [ n - 1 ] [ n - 1 ];
  if ( x < smallest || x > largest ) return - 1;
  int i = 0, j = n - 1;
  while ( i < n && j >= 0 ) {
    if ( mat [ i ] [ j ] == x ) {
      cout << "n Found at " << i << ", " << j;
      return 1;
    }
    if ( mat [ i ] [ j ] > x ) j --;
    else i ++;
  }
  cout << "n Element not found";
  return 0;
}

File: results/java_codex/SEARCH_IN_ROW_WISE_AND_COLUMN_WISE_SORTED_MATRIX.java
void search(int mat[4][4], int n, int x)
{
    int i = 0, j = n-1;  //set indexes for top right element
    while ( i < n && j >= 0 )
    {
        if ( mat[i][j] == x )
        {
            cout << "n Found at " << i << ", " << j;
            return;
        }
        if ( mat[i][j] > x )
            j--;
        else //  if mat[i][j] < x
            i++;
    }
    cout << "n Element not found";
    return;
}